pkg/checkup/workspace.go

```diff
@@ -0,0 +1,346 @@
+package checkup
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/client-go/kubernetes"
+	"log"
+	"strings"
+	"time"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	checkupNamespaceName           = "checkup-workspace"
+	resultsConfigMapName           = "checkup-results"
+	checkupJobName                 = "checkup-job"
+	serviceAccountName             = "checkup-sa"
+	resultsConfigMapWriterRoleName = "results-configmap-writer"
+)
+
+const (
+	resultsConfigMapNameEnvVarName      = "RESULT_CONFIGMAP_NAME"
+	resultsConfigMapNameEnvVarNamespace = "RESULT_CONFIGMAP_NAMESPACE"
+)
+
+const (
+	ClusterRoleKind = "ClusterRole"
+	RoleKind        = "Role"
+)
+
+type workspace struct {
+	namespace           *corev1.Namespace
+	serviceAccount      *corev1.ServiceAccount
+	resultConfigMap     *corev1.ConfigMap
+	roles               map[string]rbacv1.Role
+	roleBindings        map[string]rbacv1.RoleBinding
+	clusterRoles        map[string]rbacv1.ClusterRole
+	clusterRoleBindings map[string]rbacv1.ClusterRoleBinding
+	envVars             []corev1.EnvVar
+	resources           corev1.ResourceList
+	checkupTimeout      time.Duration
```

Comment:
This can be taken from the spec if it was here.

Reply:
Right but they you would have spec just for that, so I rather take only what needed.